pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -174,34 +174,33 @@ def __init__(self, path: str) -> None:
         self.path = path
         self.ts = None
         self._pts = None
-        self._is_valid = self.check_valid()
+        self._is_valid = None  # calculated on demand
 
     @property
     def is_valid(self):
+        if self._is_valid is None:
+            self._is_valid = self.load_valid_container() is not None
         return self._is_valid
```

Comment:
Read the other comments first!
...
This becomes then e.g.
```python
def is_valid(self):
    # check cached state
    if self._is_valid is None:
        try:
            self.load_container()
        except InvalidContainerError:
            self._is_valid = False
        else:
            self.is_valid = True
    return self._is_valid
```
which might be a bit more verbose, but reads so much clearer in what's going on here.

Reply:
I agree, although I would move the `self._is_valid = True` into the try-block

Reply:
Totally fine with that.